### PR TITLE
Lfo ring buffer

### PR DESCRIPTION
**Pull request by likelian**

### CODE REVIEW

File: src/Tests/Tests/Test_Vibrato.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+    void TearDown() override
+    {
+        for (int c = 0; c < numChannels; c++)
+        {
+            delete [] inputBuffer[c];
+            delete [] outputBuffer[c];
+        }
+
+        delete [] inputBuffer;
```
Comment (still18) [inline src/Tests/Tests/Test_Vibrato.cpp:39]: Set these equal to 0 (for standard practice, functionally it's probably fine)

File: src/Tests/Tests/Test_Vibrato.cpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    int blockSizes [] = {1024, 1025, 3, -1, 2222};
+    for (int i = 0; i < sizeof(blockSizes)/sizeof(blockSizes[0]); i++)
+    {
+        //set delay to 0.1s
+        vibrato->process(inputBuffer, outputBuffer, numSamples);
+
+
+    }
+
+
```
Comment (still18) [inline src/Tests/Tests/Test_Vibrato.cpp:129]: A lot of extra lines here. Just a design thing worth mentioning, nothing wrong with it haha

File: src/Tests/Tests/Test_Vibrato.cpp [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    }
+}
+
+/*
+ One or more additional test(s) to verify other expected behaviors.
+ */
+TEST_F(Vibrato, OneMoreTest)
+{
+
```
Comment (still18) [inline src/Tests/Tests/Test_Vibrato.cpp:169]: Possible tests could involve LFO, incorrect values, handling errors

File: src/Tests/Tests/RingBufferTest.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Setup ring buffer with values from 0 - 9
+    for(int i = 0; i < 10; i++)
+    {
+        test->putPostInc(i+1);
+    }
+
+    EXPECT_TRUE(test->get(Offset) == 8.25);
+
+}
```
Comment (still18) [inline src/Tests/Tests/RingBufferTest.cpp:66]: More tests could be provided for "too large" or negative indexes

**Review by still18 — COMMENTED** [review]
Inline comments should be provided, but overall everything looks good. Code is pretty clean and everything seems functional. More tests could be beneficial but otherwise no major issues

File: src/Tests/Tests/LfoTest.cpp [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    float fTolerance = 0.01;
+
+    std::ofstream outFile;
+    outFile.open("sineOutNikhil.txt");
+    
+    for (int i = 0; i < inNumSamplesToRender; i++) {
+
+        outFile << buffer->getPostInc(0)<<"\n";
+      
+//      EXPECT_NEAR(buffer[i], sineBuffer[i], fTolerance) << "Vectors buffer and sineBuffer differ at index " << i;
```
Comment (hchen605) [inline src/Tests/Tests/LfoTest.cpp:77]: So it is like you write out a file and compare externally? Is there any issue with EXPECT_NEAR?

File: src/Tests/Tests/RingBufferTest.cpp (hunk 4 above)
Comment (hchen605) [inline src/Tests/Tests/RingBufferTest.cpp:66]: Also maybe you can check if read behavior before write.

File: src/Tests/Tests/Test_Vibrato.cpp [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        }
+
+
+    }
+}
+
+/*
+ One or more additional test(s) to verify other expected behaviors.
+ */
+TEST_F(Vibrato, OneMoreTest)
```
Comment (hchen605) [inline src/Tests/Tests/Test_Vibrato.cpp:167]: We check if the difference of input and output is bounded. Just to avoid any unwanted glitch. Also a bit double check the fractional ringbuffer.

File: src/Tests/Tests/Test_Vibrato.cpp [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+ */
+TEST_F(Vibrato, checkBlock)
+{
+    float fSampleRateInHz = 16000.;
+    float fAmp = 0.;
+    float fWidth = 0.1;
+    vibrato->init(fWidth, fSampleRateInHz, fAmp, 1., 2);
+
+    int blockSizes [] = {1024, 1025, 3, -1, 2222};
+    for (int i = 0; i < sizeof(blockSizes)/sizeof(blockSizes[0]); i++)
```
Comment (hchen605) [inline src/Tests/Tests/Test_Vibrato.cpp:121]: Not sure about this. How is the block size chosen?

**Review by hchen605 — COMMENTED** [review]
Overall, I think the code is clean and easy to review. Just a few comments about the tests. Good job!
